src/HDWallet.cpp

```diff
@@ -62,12 +70,18 @@ HDWallet::~HDWallet() {
     std::fill(passphrase.begin(), passphrase.end(), 0);
 }
 
-void HDWallet::updateEntropy() {
-    // generate entropy (from mnemonic)
-    Data entropyRaw(32 + 1);
-    auto entropyBits = mnemonic_to_bits(mnemonic.c_str(), entropyRaw.data());
+void HDWallet::updateSeedAndEntropy() {
+    // generate seed from mnemonic
+    // it is assumed that mnemonic is valid, enforced before calling
+    mnemonic_to_seed(mnemonic.c_str(), passphrase.c_str(), seed.data(), nullptr);
+
+    // generate entropy from mnemonic
+    Data entropyRaw((Mnemonic::MaxWords * 11) / 8);
+    auto entropyBytes = mnemonic_to_bits(mnemonic.c_str(), entropyRaw.data()) / 8;
+    assert(entropyBytes <= ((Mnemonic::MaxWords * 11) / 8) && entropyBytes >= ((Mnemonic::MinWords * 11) / 8));
```

Comment:
where do you check that imported number of words <= MaxWords

Reply:
That's checked inside the c'tor with mnemonic, which calls Mnemonic::isValid(), which calls mnemonic_check().

https://github.com/trustwallet/wallet-core/blob/ar/hdwallet-error-fix/src/HDWallet.cpp#L49
https://github.com/trustwallet/wallet-core/blob/ar/hdwallet-error-fix/src/Mnemonic.cpp#L21
https://github.com/trustwallet/wallet-core/blob/ar/hdwallet-error-fix/trezor-crypto/crypto/bip39.c#L169